controllers/components/modelregistry/modelregistry_controller_actions.go

```diff
@@ -7,42 +7,49 @@ import (
 
 	operatorv1 "github.com/openshift/api/operator/v1"
 	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	"github.com/opendatahub-io/opendatahub-operator/v2/apis/common"
 	componentApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1alpha1"
 	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
-	odherrors "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/errors"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/conditions"
 	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
 	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 
 	_ "embed"
 )
 
-func checkPreConditions(_ context.Context, rr *odhtypes.ReconciliationRequest) error {
-	mr, ok := rr.Instance.(*componentApi.ModelRegistry)
-	if !ok {
-		return fmt.Errorf("resource instance %v is not a componentApi.ModelRegistry", rr.Instance)
-	}
+func checkPreConditions(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Conditions.MarkTrue(status.ConditionServerlessAvailable)
```

Comment:
i think modelreg does not care about serverless,
do we need this condition on modelreg ?

Reply:
looks like a copy and paste error,  it should be `ConditionServiceMeshAvailable`